### DISCUSSION TITLE
Will OperationDefaultValues still be required in .NET 9.0?

### DISCUSSION THREAD

**Discussion by marcominerva** [dotnet/aspnetcore#57837] [Q&A] [answered]
I'm analyzing the [OpenApiOptionsExtensions.cs](https://github.com/dotnet/eShop/blob/release/9.0/src/eShop.ServiceDefaults/OpenApiOptionsExtensions.cs) file from **release/9.0** branch of [eShop](https://github.com/dotnet/eShop).

In particular, I see the [ApplyOperationDefaultValues](https://github.com/dotnet/eShop/blob/d4614c91a99c0e6c43f2b58d1388f3386bcffe85/src/eShop.ServiceDefaults/OpenApiOptionsExtensions.cs#L141-L192) method, that is similar to the one that was typically used with [Swashbuckle.AspNetCore](https://github.com/domaindrivendev/Swashbuckle.AspNetCore):

```csharp
public static OpenApiOptions ApplyOperationDefaultValues(this OpenApiOptions options)
{
    options.AddOperationTransformer((operation, context, cancellationToken) =>
    {
        var apiDescription = context.Description;

        operation.Deprecated |= apiDescription.IsDeprecated();

        // remove any assumed media types not present in the api description
        foreach (var responseType in context.Description.SupportedResponseTypes)
        {
```

I'm wondering if something like this `ApplyOperationDefaultValues` method will still be required when .NET 9.0 ships in GA.

**Comment by captainsafia** [comment, marked as answer]
@marcominerva Thanks for reminding me that I actually need to make some updates to this branch before we merge. We shouldn't be using the `ApplyOperationDefaultValues` as is in this example. The only behavior that we need to retain is support for setting the `Deprecated` property based on the `IsDeprecated` method. That's because we don't really have a good strategy for documenting deprecation status in ApiExplorer yet. An issue was recently opened about this in the dotnet/aspnetcore repo that covers what we might need to do to be able to get rid of this workaround. See https://github.com/dotnet/aspnetcore/issues/57853 for more info.

The other workarounds that are used here can be removed. I'll update the branch.
